options/options.go

```diff
@@ -2213,16 +2216,38 @@ func (interp *interpreter) messageLiteralValue(mc *internal.MessageContext, fiel
 			if typeURLDescriptor == nil || typeURLDescriptor.Kind() != protoreflect.StringKind {
 				return interpretedFieldValue{}, reporter.Errorf(interp.nodeInfo(fieldNode.Name), "%vfailed to set type_url string field on Any: %w", mc, err)
 			}
-			msg.Set(typeURLDescriptor, protoreflect.ValueOfString(fullURL))
+			typeURLVal := protoreflect.ValueOfString(fullURL)
+			msg.Set(typeURLDescriptor, typeURLVal)
 			valueDescriptor := fmd.Fields().ByNumber(2)
 			if valueDescriptor == nil || valueDescriptor.Kind() != protoreflect.BytesKind {
 				return interpretedFieldValue{}, reporter.Errorf(interp.nodeInfo(fieldNode.Name), "%vfailed to set value bytes field on Any: %w", mc, err)
 			}
-			b, err := proto.MarshalOptions{Deterministic: true}.Marshal(msgVal.val.Message().Interface())
+
+			b, err := appendOptionBytes(nil, msgVal.msgVal)
 			if err != nil {
 				return interpretedFieldValue{}, reporter.Errorf(interp.nodeInfo(fieldNode.Val), "%vfailed to serialize message value: %w", mc, err)
 			}
 			msg.Set(valueDescriptor, protoreflect.ValueOfBytes(b))
+			msgVal.preserialized = asMessageBytes(2, b) // no need to recompute this for canonical option bytes
+			flds = []*interpretedField{
+				{
+					node:   fieldNode.Name,
+					number: 1,
+					kind:   protoreflect.StringKind,
+					value: interpretedFieldValue{
+						val: typeURLVal,
+					},
+				},
+				{
+					node:   fieldNode.Val,
+					number: 2,
+					// Technically this field is a "bytes" kind. But the actual
+					// byte output is the same, and this way we can defer the
+					// computation of those bytes until later if needed.
+					kind:  protoreflect.MessageKind,
+					value: msgVal,
+				},
+			}
```

Comment:
🤦. This was a total brainfart that this condition didn't previously return any fields. This is why expanded `Any` messages were just _absent_ from the canonical bytes output.